x-pack/metricbeat/module/aws/utils.go

```diff
@@ -0,0 +1,151 @@
+	getMetricDataInput := &cloudwatch.GetMetricDataInput{
+		NextToken:         nextToken,
+		StartTime:         &startTime,
+		EndTime:           &endTime,
+		MetricDataQueries: metricDataQueries,
+	}
+
+	reqGetMetricData := svc.GetMetricDataRequest(getMetricDataInput)
+	getMetricDataOutput, err := reqGetMetricData.Send()
+	if err != nil {
+		err = errors.Wrap(err, "Error GetMetricDataInput")
+		return nil, err
+	}
+	return getMetricDataOutput, nil
+}
+
+// GetMetricDataResults function uses MetricDataQueries to get metric data output.
+func GetMetricDataResults(metricDataQueries []cloudwatch.MetricDataQuery, svc cloudwatchiface.CloudWatchAPI, startTime time.Time, endTime time.Time) ([]cloudwatch.MetricDataResult, error) {
+	init := true
+	getMetricDataOutput := &cloudwatch.GetMetricDataOutput{NextToken: nil}
+	for init || getMetricDataOutput.NextToken != nil {
+		init = false
+		output, err := getMetricDataPerRegion(metricDataQueries, getMetricDataOutput.NextToken, svc, startTime, endTime)
+		if err != nil {
+			err = errors.Wrap(err, "getMetricDataPerRegion failed")
+			return getMetricDataOutput.MetricDataResults, err
+		}
+		getMetricDataOutput.MetricDataResults = append(getMetricDataOutput.MetricDataResults, output.MetricDataResults...)
+	}
+	return getMetricDataOutput.MetricDataResults, nil
+}
+
+// EventMapping maps data in input to a predefined schema.
+func EventMapping(input map[string]interface{}, schema s.Schema) (common.MapStr, error) {
+	return schema.Apply(input, s.FailOnRequired)
+}
+
+// GetConfigForTest function gets aws credentials for integration tests.
+func GetConfigForTest(metricSetName string) (map[string]interface{}, string) {
+	accessKeyID, okAccessKeyID := os.LookupEnv("AWS_ACCESS_KEY_ID")
+	secretAccessKey, okSecretAccessKey := os.LookupEnv("AWS_SECRET_ACCESS_KEY")
+	sessionToken, okSessionToken := os.LookupEnv("AWS_SESSION_TOKEN")
+	defaultRegion, _ := os.LookupEnv("AWS_REGION")
+	if defaultRegion == "" {
+		defaultRegion = "us-west-1"
+	}
+
+	info := ""
+	config := map[string]interface{}{}
+	if !okAccessKeyID || accessKeyID == "" {
+		info = "Skipping TestFetch; $AWS_ACCESS_KEY_ID not set or set to empty"
+	} else if !okSecretAccessKey || secretAccessKey == "" {
+		info = "Skipping TestFetch; $AWS_SECRET_ACCESS_KEY not set or set to empty"
+	} else {
+		config = map[string]interface{}{
+			"module":            "aws",
+			"period":            "300s",
+			"metricsets":        []string{metricSetName},
+			"access_key_id":     accessKeyID,
+			"secret_access_key": secretAccessKey,
+			"default_region":    defaultRegion,
+		}
+
+		if okSessionToken && sessionToken != "" {
+			config["session_token"] = sessionToken
+		}
+	}
+	return config, info
+}
+
+// CheckEventField function checks a given field type and compares it with the expected type for integration tests.
+func CheckEventField(metricName string, expectedType string, event mb.Event, t *testing.T) {
```

Comment:
Move it to some test file so we don't have to include `testing` and `assert` in non-test files. 

Also on this kind of helpers `t *testing.T` use to be the first argument.
And you can consider marking the function as being a helper by calling to [`t.Helper()`](https://golang.org/pkg/testing/#B.Helper). Testing framework takes this into account when printing errors.

Reply:
@jsoriano You are right! Thanks for pointing this out to me. I moved this function into a separate file so non-testing files are clean now. https://github.com/elastic/beats/blob/45e0ea88618ce7b441384bfa03bda3e79cc02d88/x-pack/metricbeat/module/aws/aws_testing.go

Reply:
Use the test suffix for test files, like `aws_test.go`.

Reply:
@jsoriano I ended up creating `mtest` and put the test file there. Thank you for the suggestion! Please let me know if it looks good now :-)

Reply:
Thanks for changing this.